lib/literal/array.rb

```diff
@@ -196,6 +218,24 @@ def reject!(...)
 		self
 	end
 
+	def replace(value)
```

Comment:
I'm wondering if this should work more like map where you could say replace a `Literal::Array(Integer)` with a `Literal::Array(String)` ? 🤔

Reply:
I think we should avoid any type-mutating mappings. We should be able to consider the type of a Literal::Array frozen even if the contents are mutable (within the constraints of the type).

We can skip checking types on a `Literal::Array(@__type__)` here since it’s already the same type.

Reply:
To clarify my comment, in the issue I see following example code
```ruby
array = Literal::Array(Integer).new(1, 2, 3)
array.map(String, &:to_s) # ["1", "2", "3", "4"]
```
So that's kind of where I was getting this notion of type mutating from (when that new type is specified). But if you're saying we don't want to do that here, that's fine by me 👍

Reply:
I like this for anything that returns a new object, but not for things that mutate the object. I think it’s useful to know if you’ve got a Literal::Array(Integer), it’s never going to turn into a Literal::Array(String), even if things might be added or removed form it.